### DISCUSSION TITLE
Implement GetCurrentClientType() through the "Client Type Management" codeunit

### DISCUSSION THREAD

**Discussion by Arthurvdv** [StefanMaron/BusinessCentral.LinterCop#743] [New Rule]
The idea of the rule would be to suggest to apply the "Client Type Management" codeunit to retrieve the `CurrentClientType()`.

### Example
```AL
if Session.CurrentClientType::Api <> Session.CurrentClientType then
   // Some Business Logic Here
```

### Improved
```AL
if ClientTypeManagement.GetCurrentClientType() <> CurrentClientType::Api
   // Some Business Logic Here
```

Currently I see the benefit that this unlocks the possibility to create Automate Testing. Not sure if there are any additional benefits of using this codeunit from the System Application.

```AL
codeunit 4032 "Client Type Mgt. Impl."
{
    Access = Internal;
    SingleInstance = true;
    InherentEntitlements = X;
    InherentPermissions = X;

    procedure GetCurrentClientType() CurrClientType: ClientType
    begin
        CurrClientType := CurrentClientType();
        OnAfterGetCurrentClientType(CurrClientType);
    end;

    /// <summary>
    /// An event that indicates that subscribers should set the client type that should be returned when the GetCurrentClientType is called.
    /// </summary>
    /// <remarks>
    /// Subscribe to this event from tests if you need to verify a different flow.
    /// This feature is for testing and is subject to a different SLA than production features.
    /// Do not use this event in a production environment. This should be subscribed to only in tests.
    /// </remarks>
    [InternalEvent(false)]
    local procedure OnAfterGetCurrentClientType(var CurrClientType: ClientType)
    begin
    end;
}
```
https://github.com/microsoft/BCApps/blob/137f3729a59f8814bd1b8053f048be7c57e2aac3/src/System%20Application/App/Client%20Type%20Management/src/ClientTypeMgtImpl.Codeunit.al#L29-L32

**Comment by jwikman** [comment]
The `OnAfterGetCurrentClientType` event is Internal only, so I don't think this is useful for any other project that he SystemApp project... 
Or am I missing something?

**Reply by Arthurvdv** [reply]
Oops, I've missed that 🫣

..in that case the "Client Type Management" codeunit doesn't bring anything to the table. I'm going close this one.
